src/Compilers/CSharp/Portable/Symbols/ParameterSymbol.cs

```diff
@@ -232,6 +232,15 @@ public bool HasExplicitDefaultValue
             }
         }
 
+        /// <summary>
+        /// Returns the bound default value syntax from a parameter, if it exists.
+        /// Note that this method will only return a non-null value if the 
+        /// default value was supplied in syntax and not in metadata. Thus, there are cases
+        /// where ExplicitDefaultValue will be non-null but this method will return null.
+        /// However, if ExplicitDefaultValue is null, this method should always return null.
+        /// </summary>
+        public virtual BoundParameterEqualsValue? BindParameterEqualsValue() => null;
```

Comment:
It's going to be a bit of a pain, but we try not to put virtual methods at the root here, and instead force implementations to think about whether they should be providing this method. For example, I would expect `RetargetingParameterSymbol` to always return null, while `WrappedParameterSymbol` should probably forward to the underlying parameter.

Reply:
My recommendation is to not put anything in this base type. It looks like we are only interested in this value when we are dealing with ```SourceComplexParameterSymbolBase```. I suggest to leave the API available only there and do a type check at the use site.

Reply:
From what I understand this is the case. I moved the definition out of the base type and into `SourceComplexParameterSymbolBase` and added a type check at the use site in the definite assignment pass. Later on if we do need this functionality for other parameter types we could always do the work to lift the definition into the base type and fill out all the corresponding parameter symbol implementations